float32[] v_traj
float32[] w_traj
